src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,168 @@
+#include "mesh/Mesh.hpp"
+
+namespace precice {
+
+namespace mapping {
+
+RadialGeoMultiscaleMapping::RadialGeoMultiscaleMapping(
+    Constraint     constraint,
+    int            dimensions,
+    MultiscaleType type,
+    RadialAxis     axis)
+    : Mapping(constraint, dimensions),
+      _type(type), _axis(axis)
+{
+  setInputRequirement(Mapping::MeshRequirement::VERTEX);
+  setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+}
+
+void RadialGeoMultiscaleMapping::computeMapping()
+{
+  PRECICE_TRACE(output()->vertices().size());
+
+  PRECICE_ASSERT(input().get() != nullptr);
+  PRECICE_ASSERT(output().get() != nullptr);
+
+  if (getConstraint() == CONSISTENT) {
+    PRECICE_DEBUG("Compute consistent mapping");
+  } else {
+    PRECICE_ASSERT(getConstraint() == CONSERVATIVE, getConstraint());
+    PRECICE_ASSERT(false, "Not yet implemented");
+    PRECICE_DEBUG("Compute conservative mapping");
+  }
+  _hasComputedMapping = true;
+}
+
+void RadialGeoMultiscaleMapping::clear()
+{
+  PRECICE_TRACE();
+  _hasComputedMapping = false;
+}
+
+void RadialGeoMultiscaleMapping::mapConservative(DataID inputDataID, DataID outputDataID)
+{
+  PRECICE_ASSERT(getConstraint() == CONSERVATIVE, getConstraint());
+  PRECICE_ASSERT(false, "Not yet implemented");
+  PRECICE_DEBUG("Map conservative");
+}
+
+void RadialGeoMultiscaleMapping::mapConsistent(DataID inputDataID, DataID outputDataID)
+{
+  PRECICE_TRACE(inputDataID, outputDataID);
+
+  const Eigen::VectorXd &inputValues     = input()->data(inputDataID)->values();
+  int                    valueDimensions = input()->data(inputDataID)->getDimensions();
+  Eigen::VectorXd &      outputValues    = output()->data(outputDataID)->values();
+
+  int coord = -1;
+  if (_axis == X) {
+    coord = 0;
+  } else if (_axis == Y) {
+    coord = 1;
+  } else if (_axis == Z) {
+    coord = 2;
+  } else {
+    PRECICE_ASSERT(false, "Unknown axis.");
+  }
+
+  PRECICE_ASSERT((inputValues.size() / valueDimensions == static_cast<int>(input()->vertices().size())),
+                 inputValues.size(), valueDimensions, input()->vertices().size());
+  PRECICE_ASSERT((outputValues.size() / valueDimensions == static_cast<int>(output()->vertices().size())),
+                 outputValues.size(), valueDimensions, output()->vertices().size());
+  PRECICE_ASSERT(valueDimensions == 1);
+
+  PRECICE_DEBUG("Map consistent");
+  if (_type == SPREAD) {
+    size_t const inSize  = input()->vertices().size();
+    size_t const outSize = output()->vertices().size();
+
+    Eigen::VectorXd axisMidpoints(inSize);
+    for (size_t i = 0; i < (inSize - 1); i++) {
+      auto axisPositionCurrent = input()->vertices()[i].getCoords()[coord];
+      auto axisPositionNext    = input()->vertices()[i + 1].getCoords()[coord];
+      axisMidpoints(i)         = (axisPositionCurrent + axisPositionNext) / 2;
+    }
+    axisMidpoints(inSize - 1) = 1e100; // arbitrary large number, such that vertices after the last midpoint are still assigned
```

Comment:
You can also use some C++ constant from `numeric_limits` instead of an arbitrary number.